src/posit/connect/content.py

```diff
@@ -137,12 +140,231 @@ class ContentItem(Resource):
         Tags associated with the content item.
     """
 
+    # CRUD Methods
+
+    def delete(self) -> None:
+        """Delete the content item."""
+        path = f"v1/content/{self.guid}"
+        url = urls.append(self.config.url, path)
+        self.session.delete(url)
+
+    def deploy(self) -> tasks.Task:
+        """Deploy the content.
+
+        Spawns an asynchronous task, which activates the latest bundle.
+
+        Returns
+        -------
+        tasks.Task
+            The task for the deployment.
+
+        Examples
+        --------
+        >>> task = content.deploy()
+        >>> task.wait_for()
+        None
+        """
+        path = f"v1/content/{self.guid}/deploy"
+        url = urls.append(self.config.url, path)
+        response = self.session.post(url, json={"bundle_id": None})
+        result = response.json()
+        ts = tasks.Tasks(self.config, self.session)
+        return ts.get(result["task_id"])
+
+    def refresh(self) -> Task | None:
```

Comment:
I still feel like `refresh` and `restart` don't quite map well onto the problem space for me, but also like I haven't found the right way to explain how I'm thinking about it. Gonna take this opportunity to take another run at it, but don't feel like this is a comment on the PR itself.

We wanted `refresh` to be a method that's a good catch-all term that works generically for all content types.

I think we included `restart` because we reasoned that people with interactive content types might look for that verb.

I think similarly R Markdown and Quarto users will look for a `render` verb — that's definitely the preeminent term in that world (see the second heading of the Quarto [Getting Started](https://quarto.org/docs/get-started/hello/rstudio.html#rendering) guide).

I'm fine using `refresh` as a catch-all term, but I think it feels to me like we should _either_ have both `restart` and `render`, or have neither. I think it's just having one of them which feels off to me. 🤔